mypy crashes with KeyError: 'django_stubs_ext' when using django-stubs 1.9.0

# Bug report

mypy crashes when reaching code with Django queries that annotate fields, which did not happen which django-stubs version 1.8.0.

## What's wrong

```
viewer/models.py:951: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.910
viewer/models.py:951: : note: use --pdb to drop into pdb
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 3911, in accept
  File "mypy/nodes.py", line 1558, in accept
  File "mypy/checkexpr.py", line 271, in visit_call_expr
  File "mypy/checkexpr.py", line 353, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 858, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 917, in check_call
  File "mypy/checkexpr.py", line 1029, in check_callable_call
  File "mypy/checkexpr.py", line 738, in apply_function_plugin
  File "/home/gitlab-runner/builds/b5691a72/0/acctest/proj/ENV/lib/python3.9/site-packages/mypy_django_plugin/transformers/querysets.py", line 231, in extract_proper_type_queryset_annotate
    annotated_type = get_or_create_annotated_type(api, model_type, fields_dict=fields_dict)
  File "/home/gitlab-runner/builds/b5691a72/0/acctest/proj/ENV/lib/python3.9/site-packages/mypy_django_plugin/transformers/models.py", line 447, in get_or_create_annotated_type
    model_module_file = api.modules[model_module_name]  # type: ignore
KeyError: 'django_stubs_ext'
```

## How is that should be

I Just upgraded from 1.8.0 to 1.9.0, and it was working without errors before upgrading.

## System information

- OS: Ubuntu 18.04.5 LTS
- `python` version: 3.9.1
- `django` version: 3.2.7
- `mypy` version: 0.910
- `django-stubs` version: 1.9.0

Previous error raises for python 3.9. For 3.10 it's different:

```
Traceback (most recent call last):
  File "/Users/melevir/.virtualenvs/test3.10/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/__main__.py", line 11, in console_entry
    main(None, sys.stdout, sys.stderr)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/main.py", line 87, in main
    res, messages, blockers = run_build(sources, options, fscache, t0, stdout, stderr)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/main.py", line 165, in run_build
    res = build.build(sources, options, None, flush_errors, fscache, stdout, stderr)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/build.py", line 179, in build
    result = _build(
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/build.py", line 254, in _build
    graph = dispatch(sources, manager, stdout)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/build.py", line 2697, in dispatch
    process_graph(graph, manager)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/build.py", line 3014, in process_graph
    process_fresh_modules(graph, prev_scc, manager)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/build.py", line 3092, in process_fresh_modules
    graph[id].fix_cross_refs()
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/build.py", line 1991, in fix_cross_refs
    fixup_module(self.tree, self.manager.modules,
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/fixup.py", line 26, in fixup_module
    node_fixer.visit_symbol_table(tree.names, tree.fullname)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/fixup.py", line 92, in visit_symbol_table
    value.node.accept(self)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/nodes.py", line 687, in accept
    return visitor.visit_func_def(self)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/fixup.py", line 100, in visit_func_def
    func.type.accept(self.type_fixer)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/types.py", line 1156, in accept
    return visitor.visit_callable_type(self)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/fixup.py", line 185, in visit_callable_type
    ct.ret_type.accept(self)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/types.py", line 846, in accept
    return visitor.visit_instance(self)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/fixup.py", line 154, in visit_instance
    inst.type = lookup_qualified_typeinfo(self.modules, type_ref, self.allow_missing)
  File "/Users/melevir/.virtualenvs/test3.10/lib/python3.10/site-packages/mypy/fixup.py", line 274, in lookup_qualified_typeinfo
    assert allow_missing, "Should never get here in normal mode," \
AssertionError: Should never get here in normal mode, got TypeAlias:django.db.models.query.QuerySet instead of TypeInfo
```

Feels like something's wrong with types of annotated queryset: without `.annotate` everything works fine even with `follow_imports = skip`.

@sobolevn I've tried to create a minimal django app to reproduce.

See https://github.com/sidmitra/django-stubs-issue

Also see the github failed github action with the mypy traceback: 
https://github.com/sidmitra/django-stubs-issue/runs/4239774289?check_suite_focus=true

```bash
./stubissue/fake_issue.py:9: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.910
./stubissue/fake_issue.py:9: : note: use --pdb to drop into pdb
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 3911, in accept
  File "mypy/nodes.py", line 1558, in accept
  File "mypy/checkexpr.py", line 271, in visit_call_expr
  File "mypy/checkexpr.py", line 353, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 858, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 917, in check_call
  File "mypy/checkexpr.py", line 1029, in check_callable_call
  File "mypy/checkexpr.py", line 738, in apply_function_plugin
  File "/tmp/venv/lib/python3.9/site-packages/mypy_django_plugin/transformers/querysets.py", line 231, in extract_proper_type_queryset_annotate
    annotated_type = get_or_create_annotated_type(api, model_type, fields_dict=fields_dict)
  File "/tmp/venv/lib/python3.9/site-packages/mypy_django_plugin/transformers/models.py", line 447, in get_or_create_annotated_type
    model_module_file = api.modules[model_module_name]  # type: ignore
KeyError: 'django_stubs_ext'
```

Note that this only happens with 1.9.0v of django-stubs, but not 1.8.0, so it's holding up an upgrade. I'll try to dig deeper this week, but not very familiar with the new django-stubs-ext stuff.

Even if type annotation with queryset annotate() + Count() is not possible, it might still be workable if i can just `#type: ignore` any false positives. Currently it just crashes on any file with Count() on a larger project.

I updated my [project to reproduce this](https://github.com/sidmitra/django-stubs-issue) with the latest mypy and git master branch for django-stubs. This error still seems to be present.


https://github.com/sidmitra/django-stubs-issue/runs/6951208566?check_suite_focus=true

```
./stubissue/fake_issue.py:9: error: INTERNAL ERROR -- Please try using mypy master on GitHub:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.961
./stubissue/fake_issue.py:9: : note: use --pdb to drop into pdb
Traceback (most recent call last):
  File "mypy/checkexpr.py", line 3991, in accept
  File "mypy/nodes.py", line 1760, in accept
  File "mypy/checkexpr.py", line 305, in visit_call_expr
  File "mypy/checkexpr.py", line 383, in visit_call_expr_inner
  File "mypy/checkexpr.py", line 892, in check_call_expr_with_callee_type
  File "mypy/checkexpr.py", line 957, in check_call
  File "mypy/checkexpr.py", line [10](https://github.com/sidmitra/django-stubs-issue/runs/6951208566?check_suite_focus=true#step:6:11)85, in check_callable_call
  File "mypy/checkexpr.py", line 772, in apply_function_plugin
  File "/tmp/venv/lib/python3.9/site-packages/mypy_django_plugin/transformers/querysets.py", line [23](https://github.com/sidmitra/django-stubs-issue/runs/6951208566?check_suite_focus=true#step:6:24)1, in extract_proper_type_queryset_annotate
    annotated_type = get_or_create_annotated_type(api, model_type, fields_dict=fields_dict)
  File "/tmp/venv/lib/python3.9/site-packages/mypy_django_plugin/transformers/models.py", line 576, in get_or_create_annotated_type
    model_module_file = api.modules[model_module_name]  # type: ignore
KeyError: 'django_stubs_ext'
Error: Process completed with exit code 2.
```

--- a/mypy_django_plugin/transformers/models.py
+++ b/mypy_django_plugin/transformers/models.py
@@ -20,7 +20,7 @@ from mypy.nodes import (
 from mypy.plugin import AnalyzeTypeContext, AttributeContext, CheckerPluginInterface, ClassDefContext
 from mypy.plugins import common
 from mypy.semanal import SemanticAnalyzer
-from mypy.types import AnyType, Instance, LiteralType, TypedDictType, TypeOfAny, TypeType, get_proper_type
+from mypy.types import AnyType, Instance, LiteralType, ProperType, TypedDictType, TypeOfAny, TypeType, get_proper_type
 from mypy.types import Type as MypyType
 from mypy.typevars import fill_typevars
 
@@ -769,7 +769,7 @@ def handle_annotated_type(ctx: AnalyzeTypeContext, django_context: DjangoContext
 
 def get_or_create_annotated_type(
     api: Union[SemanticAnalyzer, CheckerPluginInterface], model_type: Instance, fields_dict: Optional[TypedDictType]
-) -> Instance:
+) -> ProperType:
     """
 
     Get or create the type for a model for which you getting/setting any attr is allowed.
@@ -794,7 +794,9 @@ def get_or_create_annotated_type(
         cast(TypeChecker, api), model_module_name + "." + type_name
     )
     if annotated_typeinfo is None:
-        model_module_file = api.modules[model_module_name]  # type: ignore
+        model_module_file = api.modules.get(model_module_name)  # type: ignore
+        if model_module_file is None:
+            return AnyType(TypeOfAny.from_error)
 
         if isinstance(api, SemanticAnalyzer):
             annotated_model_type = api.named_type_or_none(ANY_ATTR_ALLOWED_CLASS_FULLNAME, [])

--- a/mypy_django_plugin/transformers/querysets.py
+++ b/mypy_django_plugin/transformers/querysets.py
@@ -147,9 +147,11 @@ def get_values_list_row_type(
 
 def extract_proper_type_queryset_values_list(ctx: MethodContext, django_context: DjangoContext) -> MypyType:
     # called on the Instance, returns QuerySet of something
-    assert isinstance(ctx.type, Instance)
+    if not isinstance(ctx.type, Instance):
+        return ctx.default_return_type
     default_return_type = get_proper_type(ctx.default_return_type)
-    assert isinstance(default_return_type, Instance)
+    if not isinstance(default_return_type, Instance):
+        return ctx.default_return_type
 
     model_type = _extract_model_type_from_queryset(ctx.type)
     if model_type is None:
@@ -205,9 +207,11 @@ def gather_kwargs(ctx: MethodContext) -> Optional[Dict[str, MypyType]]:
 
 def extract_proper_type_queryset_annotate(ctx: MethodContext, django_context: DjangoContext) -> MypyType:
     # called on the Instance, returns QuerySet of something
-    assert isinstance(ctx.type, Instance)
+    if not isinstance(ctx.type, Instance):
+        return ctx.default_return_type
     default_return_type = get_proper_type(ctx.default_return_type)
-    assert isinstance(default_return_type, Instance)
+    if not isinstance(default_return_type, Instance):
+        return ctx.default_return_type
 
     model_type = _extract_model_type_from_queryset(ctx.type)
     if model_type is None:
@@ -270,9 +274,11 @@ def resolve_field_lookups(lookup_exprs: Sequence[Expression], django_context: Dj
 
 def extract_proper_type_queryset_values(ctx: MethodContext, django_context: DjangoContext) -> MypyType:
     # called on QuerySet, return QuerySet of something
-    assert isinstance(ctx.type, Instance)
+    if not isinstance(ctx.type, Instance):
+        return ctx.default_return_type
     default_return_type = get_proper_type(ctx.default_return_type)
-    assert isinstance(default_return_type, Instance)
+    if not isinstance(default_return_type, Instance):
+        return ctx.default_return_type
 
     model_type = _extract_model_type_from_queryset(ctx.type)
     if model_type is None:

